apps/src/music/blockly/extensions.js

```diff
@@ -111,3 +115,34 @@ export const playMultiMutator = {
     this.updateShape_(this.extraSoundInputCount_ - 1);
   },
 };
+
+/*
+ * Extension for the effects field, which replaces the input_dummy in the block
+ * with a dynamic field_dropdown and updates the effect value dropdown
+ * based on the selected effect name.
+ */
+export const effectsFieldExtension = function () {
+  // Set the initial state when the block gets created
+  const thisBlock = this;
+  const valuesDropdown = new Blockly.FieldDropdown(function () {
+    return FIELD_EFFECTS_OPTIONS[thisBlock.getFieldValue(FIELD_EFFECTS_NAME)];
```

Comment:
If I make this change, all the Effects block in the toolbox have the default starting value "Full". It needs the function so that it can evaluate the value based on the field value of `FIELD_EFFECTS_NAME` at the time the block gets instantiated.

Reply:
Gotcha - thanks for the explanation!

Reply:
You're welcome!